### ISSUE TITLE
Unable to use the distribution packages

### ISSUE THREAD

**Issue by hamarituc** [sous-chefs/postgresql#709]
## :ghost: Brief Description

Before version 11 of the cookbook, there was a `setup_repo` property to choose whether to use upstream or distribution packages of PostgreSQL. This property was transformed to the `source` property. The new property doesn't allow to use distribution packages anymore.

## :pancakes: Cookbook version

11.0.1

## :woman_cook:  Chef-Infra Version

17.10.0

## :tophat: Platform details

Debian 11

## Steps To Reproduce

1. Consider the following recipe

```ruby
postgresql_install 'postgresql' do
  version 13
  source :nil
  action :install
end
```

2. Converge

```
       Cinc Client, version 17.10.0
       Patents: https://www.chef.io/patents
       Infra Phase starting
       Creating a new client identity for test-debian-11 using the validator key.
       Resolving cookbooks for run list: ["apt::default", "test::test"]
       Synchronizing cookbooks:
         - apt (7.4.3)
         - postgresql (11.0.1)
         - test (1.0.0)
         - yum (7.4.1)
       Installing cookbook gem dependencies:
       Compiling cookbooks...
       Loading Cinc Auditor profile files:
       Loading Cinc Auditor input files:
       Loading Cinc Auditor waiver files:
       
       ================================================================================
       Recipe Compile Error in /tmp/kitchen/cache/cookbooks/test/recipes/test.rb
       ================================================================================
       
       Chef::Exceptions::ValidationFailed
       ----------------------------------
       Option source must be equal to one of: :repo!  You passed :nil.
       
       Cookbook Trace: (most recent call first)
       ----------------------------------------
         /tmp/kitchen/cache/cookbooks/test/recipes/test.rb:3:in `block in from_file'
         /tmp/kitchen/cache/cookbooks/test/recipes/test.rb:1:in `from_file'
       
       Relevant File Content:
       ----------------------
       /tmp/kitchen/cache/cookbooks/test/recipes/test.rb:
       
         1:  postgresql_install 'postgresql' do
         2:    version 13
         3>>   source :nil
         4:    action :install
         5:  end
       
       System Info:
       ------------
       chef_version=17.10.0
       platform=debian
       platform_version=11
       ruby=ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]
       program_name=/opt/cinc/bin/cinc-client
       executable=/opt/cinc/bin/cinc-client
       
       
       Running handlers:
       [2022-12-14T17:36:08+01:00] ERROR: Running exception handlers
       Running handlers complete
       [2022-12-14T17:36:08+01:00] ERROR: Exception handlers complete
       Infra Phase failed. 0 resources updated in 03 seconds
       [2022-12-14T17:36:08+01:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/cinc-stacktrace.out
       [2022-12-14T17:36:08+01:00] FATAL: ---------------------------------------------------------------------------------------
       [2022-12-14T17:36:08+01:00] FATAL: PLEASE PROVIDE THE CONTENTS OF THE stacktrace.out FILE (above) IF YOU FILE A BUG REPORT
       [2022-12-14T17:36:08+01:00] FATAL: ---------------------------------------------------------------------------------------
       [2022-12-14T17:36:08+01:00] FATAL: Chef::Exceptions::ValidationFailed: Option source must be equal to one of: :repo!  You passed :nil.
```

## :police_car: Expected behavior

Other values than `:repo` should be allowed for the `source`-property.

## :heavy_plus_sign: Additional context

The presence and usage of the `install_method_repo?`-method inside the `postgresql_install`-resource indicates that using the distributions packages is supported.

```ruby
def install_method_repo?
  new_resource.source.eql?(:repo)
end
```

But there is no possibility to trigger this condition as `:repo` is the default as well as the only allowed values.

```ruby
property :source, [String, Symbol],
          default: :repo,
          coerce: proc { |p| p.to_sym },
          equal_to: %i(repo),
          description: 'Installation source'
```

An other value (e.g. `:distro`) should be introduced to reintroduce the `setup_repo`-behavior. A test should be introduced to cover this case.

**Comment by hamarituc** [comment]
I would like to ask about the status of this issue. I noticed there is a branch `os-package-support` (https://github.com/sous-chefs/postgresql/tree/os-package-support) which seems to be a competing approach to #710. What's missing to get this branch merged? Is there a chance to assist in resolving this issue?

**Comment by ramereth** [comment]
@hamarituc I believe we're waiting on someone to make a PR which properly fixes it. All of us are volunteers and only have limited time to work on this. Feel free to offer a new PR which fixes this properly!

**Comment by hamarituc** [comment]
> @hamarituc I believe we're waiting on someone to make a PR which properly fixes it. All of us are volunteers and only have limited time to work on this. Feel free to offer a new PR which fixes this properly!

I submitted a pull request based on @bmhughes work in the `os-package-support` branch. From my point of view the patch solves this issue.  I just added a couple of test cases. I couldn't identify any other issues or topics left open in the patch. So I hope it's OK to do so, although I'm no the author of the patch.
